graph.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <list>
+#include <map>
+using namespace std;
+
+
+class Node{
+    public:
+        int nodeID;
+        // Placeholder for 3D location info as I'm not sure how Godot handles it
+        tuple <double, double, double> location;
+        // Placeholder for formula data
+        int data;
+        list<Node*> neighbors;
+
+        Node(){
+            nodeID = 0;
+            location = make_tuple(0,0,0);
+            data = 0; 
+        }
+        Node(int nodeID, tuple<double, double, double> location, int data){
+            this->nodeID = nodeID;
+            this->location = location;
+            this->data = data;
+        }
+};
+
+class ProofGraph{
+    int nodeCount;
+    map<int, Node> nodeMap;
```

Comment:
Use `private:` rather than just leaving them floating as implicitly private outside of public.  

```suggestion
    private:
        int nodeCount;
        map<int, Node> nodeMap;
```